src/Controller/Component/ModulesComponent.php

```diff
@@ -222,29 +222,78 @@ public function upload(array &$requestData) : void
             throw new \RuntimeException('Invalid form data: model-type');
         }
 
+        // has another stream? drop it
+        $this->removeStream($requestData);
+
         // upload file
         $filename = $requestData['file']['name'];
         $filepath = $requestData['file']['tmp_name'];
         $headers = ['Content-Type' => $requestData['file']['type']];
         $apiClient = ApiClientProvider::getApiClient();
         $response = $apiClient->upload($filename, $filepath, $headers);
 
-        // create media from stream
+        // assoc stream to media
         $streamId = $response['data']['id'];
+        $requestData['id'] = $this->assocStreamToMedia($streamId, $requestData, $filename);
+
+        // unset some data from request
+        unset($requestData['title'], $requestData['status'], $requestData['file']);
+    }
+
+    /**
+     * Remove a stream from a media, if any
+     *
+     * @param array $requestData The request data from form
+     * @return void
+     */
+    public function removeStream(array $requestData) : void
+    {
+        if (empty($requestData['id'])) {
+            return;
+        }
+
+        $apiClient = ApiClientProvider::getApiClient();
+        $response = $apiClient->get(sprintf('/%s/%s/streams', $requestData['model-type'], $requestData['id']));
+        if (empty($response['data'])) { // no streams for media
+            return;
+        }
+        $streamId = Hash::get($response, 'data.0.id');
+        $apiClient->deleteObject($streamId, 'streams');
```

Comment:
[memo] we should change method name to `delete` in SDK, since we may also delete resources (like `streams`) that will not go in the trashcan